### ISSUE TITLE
Renamer for Images

### ISSUE THREAD

**Issue by ta0a2000t** [ta0a2000t/casuafolio#5]
Make a javascript script that reads the contents of a folder and renames images to image1, image2, ..etc
These names are what the scrollable gallery expects.
Note: it should not rename the logo. the logo should remain logo.png

this will let people just copy/paste a bunch of images into the event folder for their gallery and not have to worry about renaming each image to the expected format.

**Comment by ta0a2000t** [comment]
yes, assigned! not in the public folder though. 
Just a js function that given a folder name, should read all that is inside. (assume that all file contents are images.)

for each image, rename the image to image{i}, i from 1 to n
and change the file extension to "jpg"

return the count of images.

special case:
"logo.png"

check out this: https://www.quora.com/How-do-you-change-a-files-extension-in-JavaScript

**Comment by ta0a2000t** [comment]
@kofta999 btw, this issue is similar to this as well. you might consider https://github.com/ta0a2000t/casuafolio/issues/8

**Comment by kofta999** [comment]
> yes, assigned! not in the public folder though. Just a js function that given a folder name, should read all that is inside. (assume that all file contents are images.)
> 
> for each image, rename the image to image{i}, i from 1 to n and change the file extension to "jpg"
> 
> return the count of images.
> 
> special case: ignore "logo.png"
> 
> check out this: https://www.quora.com/How-do-you-change-a-files-extension-in-JavaScript

I'd like to ask, should I consider recursively checking every sub folder in the input folder? or the input folder is guaranteed to have only Image files and no sub-folders?

**Comment by kofta999** [comment]
> @kofta999 btw, another issue is similar to this as well. you might also consider #8

then as I'd need this anyways for renaming folders, I'll make a helper function for that and export it with the rename function.

**Comment by ta0a2000t** [comment]
no sub-folders. 

I am imagining this function to be used to read each gallery's images. So for example, if you go to the public folder -> projects_images, you will see pong_replica.

your function should get "./projects_images/pong_replica/" as input and return 1

**Comment by ta0a2000t** [comment]
YES!

chat gpt made this! :)


### Step-by-Step Guide for Your First Contribution:

1. **Fork the Repo**: Click the 'Fork' button on the top-right corner of the repository page. This will create a copy of the repo under your GitHub account.

2. **Clone Your Fork**: Clone the forked repo to your local machine. You can do this by running:
    ```
    git clone https://github.com/YOUR_USERNAME/REPO_NAME.git
    ```

3. **Set Upstream Remote**: It's a good idea to set the original repository as an "upstream" remote so you can fetch changes and keep your fork updated. Run the following command:
    ```
    git remote add upstream https://github.com/ORIGINAL_OWNER/ORIGINAL_REPOSITORY.git
    ```

4. **Create a New Branch**: Before making any changes, create a new branch for your task. This keeps things neat and easier to manage. To create and switch to a new branch, you can run:
    ```
    git checkout -b my-new-feature-branch
    ```

5. **Do Your Thing**: Now, dive into the code and start working on the task as described (in this case, the JavaScript script for image paths).

6. **Commit Your Changes**: After making your changes, commit them. Make sure to write a meaningful commit message:
    ```
    git add .
    git commit -m "Add feature to retrieve image paths from folder"
    ```

7. **Fetch and Merge Upstream Changes**: Before pushing, fetch any new changes from the original repo and merge them into your feature branch to resolve any potential conflicts:
    ```
    git fetch upstream
    git merge upstream/main
    ```

8. **Push to Your Fork**: Push your changes to your fork on GitHub:
    ```
    git push origin my-new-feature-branch
    ```

9. **Create a Pull Request (PR)**: Go to the 'Pull Requests' tab in the original repo and click 'New Pull Request'. Choose the branches you want to compare (your feature branch against the original repo's main/master branch).

10. **Review and Submit**: Add some notes about what your PR is about, what it does, etc. Once you're satisfied, click 'Create Pull Request'.

11. **Wait for Review**: Maintainers of the project will review your code. They may suggest some changes or improvements. Keep an eye out for any comments!

12. **Merge and Celebrate**: Once your PR is approved, it will be merged into the main codebase. And voilà, you've just made your first contribution to an open-source project!

**Comment by kofta999** [comment]
Hello again, I'm nearly done of reading image*.jpg function but I'd like to ask a question.
Should I assume the input directory would be an absolute path?
like for example:
```js
const images = await readImagesFromDir("/home/user/casuafolio/public/projects_images/pong_replica/");
```
or would it be a relative path? (struggling to deal with relative paths atm :P)
Thanks in advance!

**Comment by ta0a2000t** [comment]
we can't use absolute paths because it will not work when others use it.
However, in your case, you are making a generic function that takes any path in string format. So using absolute now should not be an issue.
